<robot name="flat">
  <link name="base_link"></link>
  <link
    name="base_rotn">
    <inertial>
      <origin
        xyz="-0.113 0.087 0.1"
        rpy="3.14 0 0.125" />
      <mass
        value="0.773197125339031" /> 
      <inertia
        ixx="0.00103764705393415"
        ixy="6.58182970956898E-06"
        ixz="5.54891502901135E-05"
        iyy="0.00130367193834491"
        iyz="-0.000179327606683225"
        izz="0.000348493395648461" />
    </inertial>
    <visual>
      <origin
        xyz="-0.113 0.087 0.1"
        rpy="3.14 0 0.125" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/base_rotation.STL" scale="0.001 0.001 0.001" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.898039215686275 0.917647058823529 0.929411764705882 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="-0.113 0.087 0.1"
        rpy="3.14 0 0.125" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/base_rotation.STL" scale="0.001 0.001 0.001" />
      </geometry>
    </collision>
  </link>
  <joint
    name="base_rotation"
    type="continuous">
    <origin
      xyz="0 0.19 0.12"
      rpy="0 0 0" />
    <parent
      link="base_link" />
    <child
      link="base_rotn" />
    <axis
      xyz="0 0 1" />
    <limit
      effort="0"
      velocity="0.3" />
  </joint>
  <link
    name="shoulder_links">
    <inertial>
      <origin
        xyz="0 0.21 0.155"
        rpy="0 0 0" />
      <mass
        value="0.104640684850347" />
      <inertia
        ixx="7.98235631104724E-05"
        ixy="-4.92478395888344E-19"
        ixz="4.07842412768808E-05"
        iyy="0.000925048572837215"
        iyz="-3.03056955059653E-18"
        izz="0.000973974480713755" />
    </inertial>
    <visual>
      <origin
        xyz="0 0.21 0.155"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/shoulder.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.729411764705882 0.67843137254902 0.67843137254902 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0.21 0.155"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/shoulder.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="shoulder_joint"
    type="revolute">
    <origin
      xyz="0 0.009 0.04"
      rpy="0 0 0" />
    <parent
      link="base_rotn" />
    <child
      link="shoulder_links" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-0.45"
      upper="1.12"
      effort="0"
      velocity="0.1" />
  </joint>
  <link
    name="elbow_link">
    <inertial>
      <origin
        xyz="0 0.135 -0.1275"
        rpy="-0.67 0 0" />
      <mass
        value="0.0411062379032747" />
      <inertia
        ixx="1.21837641589144E-05"
        ixy="-1.69964129768756E-18"
        ixz="1.52307939635103E-06"
        iyy="0.000132211701064061"
        iyz="5.81115462912697E-19"
        izz="0.00013982909112128" />
    </inertial>
    <visual>
      <origin
        xyz="0 0.135 -0.1275"
        rpy="-0.67 0 0" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/elbow.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.729411764705882 0.67843137254902 0.67843137254902 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0 0.135 -0.1275"
        rpy="-0.67 0 0" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/elbow.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="elbow_joint"
    type="revolute">
    <origin
      xyz="0 0.43 0.345"
      rpy="-0.67 0 0" />
    <parent
      link="shoulder_links" />
    <child
      link="elbow_link" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-3.14"
      upper="3.14"
      effort="0"
      velocity="0.1" />
  </joint>
  <link
    name="gripper">
    <inertial>
      <origin
        xyz="0.005 0.02 -0.18"
        rpy="0 0 0" />
      <mass
        value="0.0411062379032747" />
      <inertia
        ixx="1.21837641589144E-05"
        ixy="-1.69964129768756E-18"
        ixz="1.52307939635103E-06"
        iyy="0.000132211701064061"
        iyz="5.81115462912697E-19"
        izz="0.00013982909112128" />
    </inertial>
    <visual>
      <origin
        xyz="0.005 0.02 -0.18"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/gripper.stl" />
      </geometry>
      <material
        name="">
        <color
          rgba="0.729411764705882 0.67843137254902 0.67843137254902 1" />
      </material>
    </visual>
    <collision>
      <origin
        xyz="0.005 0.02 -0.18"
        rpy="0 0 0" />
      <geometry>
        <mesh
          filename="package://aspha18_urdf/meshes/gripper.stl" />
      </geometry>
    </collision>
  </link>
  <joint
    name="gripper_joint"
    type="revolute">
    <origin

      xyz="0.005 0.27 -0.25"
      rpy="0 0 0" />
    <parent
      link="elbow_link" />
    <child
      link="gripper" />
    <axis
      xyz="1 0 0" />
    <limit
      lower="-1.2"
      upper="1.9"
      effort="0"
      velocity="0.1" />
  </joint>

    <link name="end"></link>
   <joint
    name="end_joint"
    type="revolute">
    <origin

      xyz="0.005 0.02 -0.18"
      rpy="0 0 0" />
    <parent
      link="gripper" />
    <child
      link="end" />
    <axis
      xyz="1 0 0" />
  </joint>
  

  </robot>